src/System.Private.CoreLib/src/System/Array.cs

```diff
@@ -2155,8 +2192,70 @@ public static int FindLastIndex<T>(T[] array, int startIndex, int count, Predica
             return -1;
         }
 
+#if REAL_MULTIDIM_ARRAYS
+        static int s_arrayBoundsZero;
+
+        /// <summary>
+        /// Get pointer to the begining of the bounds (counts for each dim)
+        /// Works for any array type.
+        /// </summary>
+        private unsafe int* GetBoundsPtr()
+        {
+            // TODO: Make this return a ByRef. For now we assume 'this' is pinned.
+            // WARNING: LET ME REPEAT: WE ASSUME 'THIS' IS PINNED
```

Comment:
I like what you had there before better, and it was also faster -  even once we got byrefs.

(I meant my comment about diff vs. CoreCLR to be just about fixing the logic - that it should return lowerbound + count.)